ifrs17/Import/Importers.ipynb

```diff
@@ -984,35 +1029,29 @@
             "source": [
                 "Import.DefineFormat(ImportFormats.Cashflow, async (options, dataSet) => {",
                 "\n    Activity.Start();",
-                "\n    var args = (await GetArgsFromMainAsync<PartitionByReportingNodeAndPeriod>(dataSet)) with {ImportFormat = ImportFormats.Cashflow};",
-                "\n    await DataNodeFactoryAsync(dataSet, ImportFormats.Cashflow, args);",
+                "\n    var mainArgs = (await GetArgsFromMainAsync<PartitionByReportingNodeAndPeriod>(dataSet)) with {ImportFormat = ImportFormats.Cashflow};",
                 "\n    if(Activity.HasErrors()) return Activity.Finish();",
+                "\n",
+                "\n    var targetArgs = mainArgs.Scenario == null ",
+                "\n                     ? mainArgs.RepeatOnce().Concat(await DataSource.Query<PartitionByReportingNodeAndPeriod>() ",
+                "\n                               .Where(x => x.ReportingNode == mainArgs.ReportingNode && x.Year == mainArgs.Year && x.Month == mainArgs.Month && x.Scenario != null)",
+                "\n                               .Select(x => new ImportArgs(x.ReportingNode, x.Year, x.Month, default(Periodicity), x.Scenario, ImportFormats.Cashflow)).ToArrayAsync())",
+                "\n                     : mainArgs.RepeatOnce();",
+                "\n",
+                "\n    if (targetArgs.Count() > 1) ApplicationMessage.Log(Warning.ScenarioReCalculations, String.Join(\", \", targetArgs.Skip(1).Select(x => x.Scenario)));",
```

Comment:
How can you guarantee the order? Skip 1 is not what you want, but something like First(x => x.Scenario is not null).

Also you have the if already on line 1035, why do you need to count targetArgs? Issue the Warning there.

Reply:
The order is guaranteed by the construction of the query in line 1036 (using of Concat()).  In line 1041 I need to collect all Scenario to produce a string to Warning Message.   
First(x => x.Scenario is not null) would return only one Scenario

Reply:
You can also use a Where() if you need more.

If mainArgs.Scenario == null, you have mainArgs.RepeatOnce(), which will have count == 1, as you don't add anything else.

All other cases seem to have count > 1, unless the Where in 1037 is empty (can it be?). Extract this scenario query in in a variable and this is the only thing you need to build the Warning --> you don't need to Concat and then separate/Skip in the next line.

You can do all this logic inside and if where mainArgs.Scenario == null

on line 1032, you can already have an array (this RepeatOnce).